src/test/datascience/kernel-launcher/localKernelFinder.unit.test.ts

```diff
@@ -32,992 +31,19 @@ import { IPythonExtensionChecker } from '../../../client/api/types';
 import { PYTHON_LANGUAGE } from '../../../client/common/constants';
 import { arePathsSame, getOSType } from '../../common';
 import { EventEmitter, Memento, Uri } from 'vscode';
-import { IDisposable } from '../../../client/common/types';
+import { IDisposable, IExtensionContext } from '../../../client/common/types';
 import { LocalKnownPathKernelSpecFinder } from '../../../client/datascience/kernel-launcher/localKnownPathKernelSpecFinder';
 import { JupyterPaths } from '../../../client/datascience/kernel-launcher/jupyterPaths';
 import { LocalPythonAndRelatedNonPythonKernelSpecFinder } from '../../../client/datascience/kernel-launcher/localPythonAndRelatedNonPythonKernelSpecFinder';
-import {
-    areInterpreterPathsSame,
-    getInterpreterHash,
-    getNormalizedInterpreterPath
-} from '../../../client/pythonEnvironments/info/interpreter';
+import { getInterpreterHash } from '../../../client/pythonEnvironments/info/interpreter';
 import { OSType } from '../../../client/common/utils/platform';
 import { disposeAllDisposables } from '../../../client/common/helpers';
-import { KernelConnectionMetadata } from '../../../client/datascience/jupyter/kernels/types';
+import {
+    KernelConnectionMetadata,
+    LocalKernelConnectionMetadata
+} from '../../../client/datascience/jupyter/kernels/types';
 import { loadKernelSpec } from '../../../client/datascience/kernel-launcher/localKernelSpecFinderBase';
-import { traceInfoIfCI } from '../../../client/common/logger';
-
-[false, true].forEach((isWindows) => {
```

Comment:
Removed all of the old tests
@rchiodo  You might recall this is something you asked for (& obviously I agreed with that suggestion).
Basically migrate old tests to the new style.

Reply:
I'm confused. Why wouldn't most of these tests still be valid?

Reply:
They are, they are duplicates and written a different way.
You suggested that I just migrate them to the new way.

Basically I wrote tests in a different way, which I believe was easier, in the test:
* You can define all global kernle specs, all interpreters & kernlespecs that exist in each interpreter
* In the previous test this wasn't straightforward, it was slightly complicated.
* In that PR you suggested I migrate all the old tests to the new style (if the new tests covered the same scenarios)